diff --git a/find/find.c b/find/find.c
index 2c63677..862657a 100644
--- a/find/find.c
+++ b/find/find.c
@@ -1618,8 +1618,6 @@ process_dir (char *pathname, char *name, int pathlen, struct stat *statp, char *
               * of what we think we should see later.
               * If there was no change, the assignments are a no-op.
               */
-             dir_ids[dir_curr].dev = stat_buf.st_dev;
-             dir_ids[dir_curr].ino = stat_buf.st_ino;
              break;
       
            case SafeChdirFailWouldBeUnableToReturn:
