diff --git a/find/pred.c b/find/pred.c
index e2e8dfb..311b3b4 100644
--- a/find/pred.c
+++ b/find/pred.c
@@ -602,6 +602,12 @@ new_impl_pred_exec (const char *pathname,
 
       /* Actually invoke the command. */
       bc_do_exec (&execp->ctl, &execp->state);
+      if (execp->wd_for_exec && execp->wd_for_exec != initial_wd)
+        {
+          free_cwd (execp->wd_for_exec);
+          free (execp->wd_for_exec);
+          execp->wd_for_exec = NULL;
+        }
       if (WIFEXITED(execp->last_child_status))
        {
          if (0 == WEXITSTATUS(execp->last_child_status))
