diff --git a/find/pred.c b/find/pred.c
index e2e8dfb..345ede4 100644
--- a/find/pred.c
+++ b/find/pred.c
@@ -510,21 +510,25 @@ pred_empty (const char *pathname, struct stat *stat_buf, struct predicate *pred_
 static bool
 record_exec_dir (struct exec_val *execp)
 {
-  if (!execp->wd_for_exec)
+  if (!execp->state.todo) //(!execp->wd_for_exec)
     {
       /* working directory not already known, so must be a *dir variant,
         and this must be the first arg we added.   However, this may
         be -execdir foo {} \; (i.e. not multiple).  */
-      assert (!execp->state.todo);
+      //assert (!execp->state.todo);
 
       /* Record the WD. */
       execp->wd_for_exec = xmalloc (sizeof (*execp->wd_for_exec));
