diff --git a/find/parser.c b/find/parser.c
index a707546..0684291 100644
--- a/find/parser.c
+++ b/find/parser.c
@@ -1313,6 +1313,11 @@ parse_newerXY (const struct parser_table* entry, char **argv, int *arg_ptr)
           * past the test name (for most other tests, this is already done)
           */
          (*arg_ptr)++;
+
+          if ((argv == NULL) || (argv[*arg_ptr] == NULL))
+            {
+              return false;
+            }
          
          our_pred = insert_primary (entry);
