diff --git a/src/dfa.c b/src/dfa.c
index 3122e22..77766ac 100644
--- a/src/dfa.c
+++ b/src/dfa.c
@@ -468,8 +468,9 @@ fetch_wc (char const *eoferr)
 static int
 in_coll_range (char ch, char from, char to)
 {
+  assert(from <= to);
   char c[6] = { from, 0, ch, 0, to, 0 };
-  return strcoll (&c[0], &c[2]) <= 0 && 0 <= strcoll (&c[2], &c[4]);
+  return strcoll (&c[0], &c[2]) <= 0 && strcoll (&c[2], &c[4]) <= 0;
 }
 
 #ifdef MBS_SUPPORT
