diff --git a/find/parser.c b/find/parser.c
index 75bbe69..4f53c5a 100644
--- a/find/parser.c
+++ b/find/parser.c
@@ -3109,6 +3109,7 @@ parse_time (const struct parser_table* entry, char *argv[], int *arg_ptr)
   if (get_comp_type(&timearg, &comp))
     {
       /* Remember, we invert the sense of the comparison, so this tests against COMP_LT instead of COMP_GT... */
+
       if (COMP_LT == tval.kind)
        {
          uintmax_t expected = origin + (DAYSECS-1);
@@ -3122,6 +3123,7 @@ parse_time (const struct parser_table* entry, char *argv[], int *arg_ptr)
       /* We discard the value of comp here, as get_relative_timestamp
        * will set tval.kind.
        */
+       if(comp != COMP_EQ) timearg--;
     }

   if (!get_relative_timestamp(timearg, &tval, origin, DAYSECS, errmsg))
