diff --git a/find/parser.c b/find/parser.c
index 75bbe69..dbe8503 100644
--- a/find/parser.c
+++ b/find/parser.c
@@ -3096,7 +3096,7 @@ parse_time (const struct parser_table* entry, char *argv[], int *arg_ptr)
   struct predicate *our_pred;
   struct time_val tval;
   enum comparison_type comp;
-  const char *timearg;
+  const char *timearg,*timearg1;
   const char *errmsg = "arithmetic overflow while converting %s days to a number of seconds";
   time_t origin;
 
@@ -3106,10 +3106,11 @@ parse_time (const struct parser_table* entry, char *argv[], int *arg_ptr)
   /* Decide the origin by previewing the comparison type. */
   origin = options.cur_day_start;
 
-  if (get_comp_type(&timearg, &comp))
+  timearg1 = timearg; /* MEOLIC BUG FIXING */
+  if (get_comp_type(&timearg1, &comp)) /* MEOLIC BUG FIXING */
     {
       /* Remember, we invert the sense of the comparison, so this tests against COMP_LT instead of COMP_GT... */
-      if (COMP_LT == tval.kind)      
+      if (COMP_LT == tval.kind) /* MEOLIC: HERE WE SHOULD HAVE "comp" INSTEAD OF "tval.kind" !! */      
         {
           uintmax_t expected = origin + (DAYSECS-1);
           origin += (DAYSECS-1);
